src/Stryker.Core/Stryker.Core/Instrumentation/AnonymousFunctionExpressionToBodyEngine.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Stryker.Core.Instrumentation
+{
+    /// <summary>
+    /// Helper that converts a method (including operators) from expression body to statement body form (or arrow to curly braces).
+    /// </summary>
+    internal class AnonymousFunctionExpressionToBodyEngine : BaseEngine<AnonymousFunctionExpressionSyntax>
+    {
+        /// <summary>
+        /// Converts the given method (or operator) from expression to body form.
+        /// </summary>
+        /// <typeparam name="T">Specific node type</typeparam>
+        /// <param name="method">Method/operator to be converted.</param>
+        /// <returns>the converted method/operator</returns>
+        /// <remarks>returns the original node if no conversion is needed/possible</remarks>
+        public T ConvertToBody<T>(T method) where T: AnonymousFunctionExpressionSyntax
+        {
+            if (method.ExpressionBody == null || method.Block != null)
+            {
+                // no need to convert
+                return method;
+            }
+
+            var  statementLine = SyntaxFactory.ReturnStatement(method.ExpressionBody.WithLeadingTrivia(SyntaxFactory.Space));
+
+            // do we need add return to the expression body?
+            return method.WithBody(SyntaxFactory.Block(statementLine)).WithAdditionalAnnotations(Marker) as T;
+        }
+
+        protected override SyntaxNode Revert(AnonymousFunctionExpressionSyntax node)
+        {
+            // get expression
+            var expression = node.Block?.Statements[0] switch
+            {
+                ReturnStatementSyntax returnStatement => returnStatement.Expression!,
```

Comment:
We don't use nullable reference type (yet). What does the `!` add?

Reply:
it marks we know for sure the value can't be null. This brings some speed improvement, but I do it to get rid of the reminder about it 😄

Reply:
I didn't know this actually does add something when Nullable reference types is not turned on, cool!